src/mscorlib/corefx/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -10,14 +10,13 @@ namespace System.Globalization
 {
     public partial class CompareInfo
     {
-        // ICU uses a char* (UTF-8) to represent a locale name.
-        private readonly byte[] m_sortNameAsUtf8;
+        private readonly Interop.GlobalizationInterop.SafeSortHandle m_sortHandle;
 
         internal unsafe CompareInfo(CultureInfo culture)
         {
             m_name = culture.m_name;
             m_sortName = culture.SortName;
-            m_sortNameAsUtf8 = System.Text.Encoding.UTF8.GetBytes(m_sortName);
+            m_sortHandle = Interop.GlobalizationInterop.GetSortHandle(System.Text.Encoding.UTF8.GetBytes(m_sortName));
```

Comment:
What do you think about lazy loading the sort handle?  That way the finalizable object isn't created when someone has code like we do in DateTimeFormatInfo:  https://github.com/dotnet/coreclr/blob/master/src/mscorlib/corefx/System/Globalization/DateTimeFormatInfo.cs#L2194-L2211